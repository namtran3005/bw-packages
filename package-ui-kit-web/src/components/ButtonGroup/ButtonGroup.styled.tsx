import * as React from 'react';
import styled from 'styled-components';

interface RootProps {
  fullWidth?: boolean;
  align?: React.CSSProperties['textAlign'];
  style?: React.CSSProperties;
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  fullWidth,
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  align,
  ...cleanProps
}) => <div {...cleanProps} />;

const Root = styled(RootComponent)`
  display: ${(props) => (props.fullWidth ? 'flex' : 'block')};
  text-align: ${(props) => (props.align ? props.align : 'initial')};

  & > * {
    flex: 1;
    margin-right: 8px !important;

    &:last-child {
      margin-right: 0 !important;
    }
  }
`;

Root.displayName = 'ButtonGroup';
RootComponent.displayName = 'ButtonGroupComponent';

export { Root, RootProps, RootComponent };
