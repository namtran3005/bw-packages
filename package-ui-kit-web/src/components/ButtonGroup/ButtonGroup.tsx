import * as React from 'react';

import { Root, RootProps as ButtonGroupProps } from './ButtonGroup.styled';

const ButtonGroup: React.FunctionComponent<ButtonGroupProps> = (props) => (
  <Root {...props} />
);

export { ButtonGroup, ButtonGroupProps };
