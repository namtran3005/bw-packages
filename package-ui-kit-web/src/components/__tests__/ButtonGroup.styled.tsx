import * as React from 'react';
import { shallow } from 'enzyme';
import { Root, RootComponent } from '../ButtonGroup/ButtonGroup.styled';

describe('ButtonGroup', () => {
  it('should render button group component', () => {
    const wrapper = shallow(<RootComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render button group', () => {
    const wrapper = shallow(<Root />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render full width button group', () => {
    const wrapper = shallow(<Root fullWidth />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render centered button group', () => {
    const wrapper = shallow(<Root align="center" />);
    expect(wrapper).toMatchSnapshot();
  });
});
