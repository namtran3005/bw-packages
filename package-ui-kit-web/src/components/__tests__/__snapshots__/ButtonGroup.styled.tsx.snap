// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render button group 1`] = `
.c0 {
  display: block;
  text-align: initial;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px !important;
}

.c0 > *:last-child {
  margin-right: 0 !important;
}

<ButtonGroupComponent
  className="c0"
/>
`;

exports[`ButtonGroup should render button group component 1`] = `<div />`;

exports[`ButtonGroup should render centered button group 1`] = `
.c0 {
  display: block;
  text-align: center;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px !important;
}

.c0 > *:last-child {
  margin-right: 0 !important;
}

<ButtonGroupComponent
  align="center"
  className="c0"
/>
`;

exports[`ButtonGroup should render full width button group 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: initial;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px !important;
}

.c0 > *:last-child {
  margin-right: 0 !important;
}

<ButtonGroupComponent
  className="c0"
  fullWidth={true}
/>
`;
